File: Java solutions/Graphs/NoOfProvinces.java
import java.util.*;
class Solution {
    public int findCircleNum(int[][] isConnected) {

        int ROWCOL = isConnected.length;
        Set<Integer> visited = new HashSet<>();
        int count = 0;

        for(int i = 0; i < ROWCOL; i++)
        {
            if(visited.add(i))
            {
                count++;
                dfs(i,visited,isConnected);
            }
        }

        return count;  
    }

    private void dfs(int node, Set<Integer> visited, int[][] isConnected)
    {

        for(int i = 0; i < isConnected.length; i++)
        {
            if(!visited.contains(i) && isConnected[node][i] == 1)
            {
                visited.add(i);
                dfs(i,visited,isConnected);
            }
        }

    }
}

File: Python solutions/Graphs/no_of_provinces.py
from ast import List
class Solution:
    def findCircleNum(self, isConnected: List[List[int]]) -> int:

        ROWCOL = len(isConnected)
        visited = [False] * ROWCOL
        count = 0

        def dfs(node: int):

            for i in range(ROWCOL):
                if not visited[i] and isConnected[node][i] == 1:
                    visited[node] = True
                    dfs(i)
            
        
        for i in range(ROWCOL):
            if not visited[i]:
                count += 1
                dfs(i)
        
        return count